### ISSUE TITLE
Admin-ui: when you have a customsField (type "localeString") for a product there is an exception during adding a new product

### ISSUE THREAD

**Issue by alex-dwt** [vendure-ecommerce/vendure#1397]
**Describe the bug**
I added a custom field to Product. Type is 'localeString'. Also I created a UI component (textArea) for editing this field, and connected them both in vendure-config.ts (like `ui: {component: 'product-short-description-input'}`).  
Editing of existed products works well. But a problem happens when I want to add a new product (click 'New Product'). Exact after the page for adding a new product is opened, I have this error:  
```
ERROR TypeError: currentTranslation is undefined
    setCustomFieldFormValues vendure-admin-ui-core.js:20656
    setFormValues vendure-admin-ui-catalog.js:5015
    init vendure-admin-ui-core.js:20636
    RxJS 39
    Angular 7
    RxJS 3
```

**To Reproduce**
1. Add a customField (type is 'localeString') for products with a UI component for editing it.
2. Open page "NEW PRODUCT"
3. See error
4. After the error happened, the JS functionality on the page partially broken, I have to fully reload the page to keep adding Product Variants

**Expected behavior**
There is must be no errors

**Environment:**
 - @vendure/core version: 1.4.5

**Additional context**
```
setFormValues(product, languageCode) {
        const currentTranslation = findTranslation(product, languageCode);
             .....
       this.setCustomFieldFormValues(this.customFields, this.detailForm.get(['product', 'customFields']), product, currentTranslation);
              .....
}

setCustomFieldFormValues(customFields, formGroup, entity, currentTranslation) {
        var _a, _b;
        for (const fieldDef of customFields) {
            const key = fieldDef.name;
            const value = fieldDef.type === 'localeString'
                ? (_a = currentTranslation.customFields) === null || _a === void 0 ? void 0 : _a[key]
        .......
}
```
I am sure that the problem is in `(_a = currentTranslation.customFields) === null`.   
I think `currentTranslation` is empty in this situation and thus an error happened.  
And because it is only when a new product is adding, I think this problem is real (because we cannot determine currentTransaltion for an empty new product, it's null, and the error arises).  
Could you please take a look, it's very annoying when you have to refresh the page after this error

**Comment by michaelbromley** [comment]
Hi, can you please share your exact custom field config for that field, and your `product-short-description-input` code (or a minimal version that reproduces the error)?

**Comment by alex-dwt** [comment]
Hello.
Nothing special.
Vendure-config:  
```
        AdminUiPlugin.init({
            route: 'admin',
            port: 3002,
            adminUiConfig: {
                hideVendureBranding: true,
                hideVersion: true,
            },
            app: compileUiExtensions({
                outputPath: path.join(__dirname, '../admin-ui-compiled'),
                extensions: [
                    {
                        extensionPath: path.join(__dirname, '../admin-ui/ui-extensions'),
                        ngModules: [{
                            type: 'shared',
                            ngModuleFileName: 'ProductShortDescriptionInput.ts',
                            ngModuleName: 'SharedExtensionModule',
                        }],
                    }
                ],
                devMode: true,
            })
        }),

    customFields: {
        Product: [
            {
                name: 'shortDescription',
                type: 'localeString',
                nullable: false,
                defaultValue: '',
                length: 65000,
                ui: {component: 'product-short-description-input'},
            },
        ],
}
```  
Angular component:
```
@Component({
	template: `<textarea [formControl]="formControl"></textarea>`,
	styles: [`textarea {
		width: 450px;
		height: 130px;
        border: 1px solid var(--color-grey-300);
		padding: 5px;
		border-radius: 3px;
	}`]
})
export class ProductShortDescriptionInput implements FormInputComponent<CustomFieldConfig> {
	readonly: boolean;
	config: CustomFieldConfig;
	formControl: FormControl;
}

@NgModule({
	imports: [SharedModule],
	declarations: [ProductShortDescriptionInput],
	providers: [
		registerFormInputComponent('product-short-description-input', ProductShortDescriptionInput),
	]
})
export class SharedExtensionModule {
}
```

  
But, as I described initially, I think we just should verify the Angular code
```
setCustomFieldFormValues(customFields, formGroup, entity, currentTranslation) {
...
}
```
and be sure that currentTranslation cannot be set as nullable from all places of invoking this function, probably  somehow  it's a really Null when a new product is going to be created

**Comment by michaelbromley** [comment]
Thanks for the extra info. Just looked into this a bit more and it appears it is the same issue as reported in #1378 and thus should have been fixed in v1.4.6. Please update and try again to verify.
